llvm/include/llvm/IR/IntrinsicsDirectX.td

```diff
@@ -35,6 +35,9 @@ def int_dx_typedBufferLoad_checkbit
 def int_dx_typedBufferStore
     : DefaultAttrsIntrinsic<[], [llvm_any_ty, llvm_i32_ty, llvm_anyvector_ty]>;
 
+def int_dx_updateCounter
+    : DefaultAttrsIntrinsic<[], [llvm_any_ty, llvm_anyint_ty]>;
```

Comment:
is there an llvm_i8_ty that could be used here for the 2nd arg? From DXIL.td it seems this is supposed to only accept i8? I see a use of llvm_i8_ty here:   https://github.com/llvm-mirror/llvm/blob/2c4ca6832fa6b306ee6a7010bfb80a3f2596f824/include/llvm/IR/IntrinsicsX86.td#L16